### ISSUE TITLE
[BUG] 4.8.3 findOneAndUpdate throws "Cannot read property 'value' of null"

### ISSUE THREAD

**Issue by misleadingTitle** [Automattic/mongoose#4993]
After the last update to version 4.8.3 the code below gives an exception.

```js
var mongoose = require("mongoose");
mongoose.connect('mongodb://localhost:27017/test');
var db = mongoose.connection;
db.on('error', console.error.bind(console, 'connection error:'));
db.once('open', function () {
    console.log("Connected...");
});

schema = mongoose.Schema({
    _id: {type: Number, match: /^\d+/},
    description: String
});
var model = mongoose.model("installation", schema);
model.findOneAndUpdate({"_id": 140}, {"_id": 999}, {new: true}, function (err, data) {
    if (err) return console.error(err);
    console.log(data);
});
```

Results in:
```
TypeError: Cannot read property 'value' of null
    at C:\clv\test_thread_safety\node_modules\mongoose\lib\query.js:2012:27
    at C:\clv\test_thread_safety\node_modules\mongoose\lib\utils.js:512:16
    at handleCallback (C:\clv\test_thread_safety\node_modules\mongoose\node_modules\mongodb\lib\utils.js:95:56)
    at C:\clv\test_thread_safety\node_modules\mongoose\node_modules\mongodb\lib\collection.js:2485:22
    at handleCallback (C:\clv\test_thread_safety\node_modules\mongoose\node_modules\mongodb\lib\utils.js:95:56)
    at C:\clv\test_thread_safety\node_modules\mongoose\node_modules\mongodb\lib\db.js:311:20
    at C:\clv\test_thread_safety\node_modules\mongoose\node_modules\mongodb-core\lib\connection\pool.js:455:18
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```
In version 4.8.2 mongoose was returning correclty the error from mongodb when a query tries to change an immutable field:
```js
{ MongoError: After applying the update to the document {_id: 140 , ...}, the (immutable) field '_id' was found to have been altered to _id: 999
    at Function.MongoError.create (C:\clv\test_thread_safety\node_modules\mongodb-core\lib\error.js:31:11)
    at C:\clv\test_thread_safety\node_modules\mongodb-core\lib\connection\pool.js:483:72
    at authenticateStragglers (C:\clv\test_thread_safety\node_modules\mongodb-core\lib\connection\pool.js:429:16)
    at Connection.messageHandler (C:\clv\test_thread_safety\node_modules\mongodb-core\lib\connection\pool.js:463:5)
    at Socket.<anonymous> (C:\clv\test_thread_safety\node_modules\mongodb-core\lib\connection\connection.js:319:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:548:20)
  name: 'MongoError',
  message: 'After applying the update to the document {_id: 140 , ...}, the (immutable) field \'_id\' was found to have been altered to _id: 999',
  ok: 0,
  errmsg: 'After applying the update to the document {_id: 140 , ...}, the (immutable) field \'_id\' was found to have been altered to _id: 999',
  code: 66,
  codeName: 'ImmutableField' }
```

Mongodb version is 3.4.0. The document with id 140 is in the database in the correct collection (installations).

**Comment by PrestonR** [comment]
Confirmed we had this issue in our project also. Had to rollback and hard lock at 4.8.2 for now.

**Comment by varunjayaraman** [comment]
Thanks, this looks like a confirmed bug:

```javascript
const mongoose = require("mongoose");
const co = require('co');

const GITHUB_ISSUE = `gh-4993`;

exec();

function exec() {
  mongoose.connect(`mongodb://localhost:27017/${GITHUB_ISSUE}`);
  var db = mongoose.connection;

  db.on('error', console.error.bind(console, 'connection error:'));
  db.once('open', function() {
    console.log("Connected...");
  });

  schema = mongoose.Schema({
    _id: { type: Number, match: /^\d+/ },
    description: String
  });

  var model = mongoose.model("installation", schema);

  model.remove({}, function(err) {
    if (err) {
      console.error(`error removing docs fro mcollection`, err);
      return;
    }

    model.create({ _id: 140 }, function(err, doc) {
      if (err) {
        console.error(err);
        return;
      }
      model.findOneAndUpdate({ "_id": doc._id }, { "_id": 999 }, { new: true }, function(err, data) {
        if (err) return console.error(err);
        console.log(data);
      });
    })
  });

}
```

**Comment by vkarpov15** [comment]
Fixed in 4.8.4
